kernel/src/actions/deletion_vector.rs

```diff
@@ -175,6 +182,17 @@ fn read_u32(cursor: &mut Cursor<Bytes>, endian: Endian) -> DeltaResult<u32> {
     }
 }
 
+/// decode a slice into a u32
+fn slice_to_u32(buf: &[u8], endian: Endian) -> DeltaResult<u32> {
+    let array = buf
+        .try_into()
```

Comment:
It's unclear to me what this is supposed to do?

Reply:
[`u32::from_be_bytes`](https://doc.rust-lang.org/core/primitive.u32.html#method.from_be_bytes) (and `from_le_bytes`)  want a `[u8; 4]`, but this is just a slice, so `try_into` tries to convert the slice into a four element array